globus_action_provider_tools/flask/apt_blueprint.py

```diff
@@ -217,22 +177,31 @@ def wrapper() -> ViewReturn:
         print(f'Registered action run plugin "{func.__name__}"')
         return wrapper
 
-    @methdispatch
-    def _generic_action_status(self, *args, **kwargs):
-        raise NotImplemented(
-            f"No action_status function registered to handle type {type(args[0])}"
-        )
+    @overload
```

Comment:
Another new python feature for me. This is kinda a weird syntax, not that there's anything that can be done about it.

How does mypy handle things if the first argument is just listed as a `Union` type rather than adding the `overload` multi-definition kinda thing?

Reply:
That's one of the first things I tried (I really didn't like the idea of creating function stubs just for using `@overload`) but mypy apparently tries to do an AND between the types declared in the `Union` so that mypy reports errors for both `str` and `ActionStatus` typed args. The only alternative was using typing Protocols but the mypy errors they generate are opaque. I figured actionable mypy errors > neatly defined typing definitions.